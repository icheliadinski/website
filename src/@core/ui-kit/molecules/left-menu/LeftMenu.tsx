import React from 'react'
import classNames from 'classnames'
import { Icon, ICON_TYPES } from '@core/ui-kit'
import { ProfileMenu } from './ProfileMenu'
import styles from './LeftMenu.module.scss'

export interface Props {
  profileIcons?: ICON_TYPES[]
}

export const LeftMenu: React.FC<Props> = ({ profileIcons = [] }) => {
  const hamburgerClasses = classNames(styles.hamburger, styles.item)
  const downloadClasses = classNames(styles.item, styles.download)

  return (
    <aside className={styles.menu}>
      <span className={hamburgerClasses} role="button" />
      <ProfileMenu icons={profileIcons} />
      <span className={downloadClasses}>
        <Icon type={ICON_TYPES.DOWNLOAD_ICON} />
      </span>
    </aside>
  )
}
