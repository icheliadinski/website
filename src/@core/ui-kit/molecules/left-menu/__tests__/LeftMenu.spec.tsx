import React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'
import { LeftMenu } from '../LeftMenu'

describe('[LeftMenu]', () => {
  let wrapper: ShallowWrapper

  beforeEach(() => {
    wrapper = shallow(<LeftMenu />)
  })

  it('renders successfully', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
