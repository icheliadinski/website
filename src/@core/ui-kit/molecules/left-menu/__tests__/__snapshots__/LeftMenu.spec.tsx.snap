// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[LeftMenu] renders successfully 1`] = `
<aside
  className="menu"
>
  <span
    className="hamburger item"
    role="button"
  />
  <ProfileMenu
    icons={Array []}
  />
  <span
    className="item download"
  >
    <Icon
      type="download"
    />
  </span>
</aside>
`;
